x-pack/plugins/cases/public/client/helpers/group_alerts_by_rule.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CommentRequestAlertType } from '../../../common/api';
+import { CommentType, Ecs } from '../../../common';
+import { getRuleIdFromEvent } from './get_rule_id_from_event';
+import { CaseAttachments } from '../../types';
+
+type Maybe<T> = T | null;
+interface Event {
+  data: EventNonEcsData[];
+  ecs: Ecs;
+}
+interface EventNonEcsData {
+  field: string;
+  value?: Maybe<string[]>;
+}
+
+export const groupAlertsByRule = (items: Event[], owner: string): CaseAttachments => {
+  const attachmentsByRule = items.reduce<Record<string, CommentRequestAlertType>>((acc, item) => {
+    const rule = getRuleIdFromEvent(item);
+    if (!acc[rule.id]) {
+      acc[rule.id] = {
+        alertId: [],
```

Comment:
nit: `alertIds`

Reply:
the API is explicitly `alertId` in this case.